### PR TITLE
EDSC-1983: Update Leaflet to 1.3.4

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: app/assets/javascripts/modules/map/leaflet-plugins/proj.js.coffee [hunk 1]
```diff
@@ -3,83 +3,78 @@
...
+  #   projection: new Proj4Projection("+proj=laea +lat_0=-90 +lon_0=0 +x_0=0 +y_0=0 +a=6371228 +b=6371228 +units=m +no_defs")
 
 
   exports =
     epsg4326: epsg4326
     epsg3413: epsg3413
     epsg3031: epsg3031
-    epsg3408: epsg3408
-    epsg3409: epsg3409
+    # epsg3408: epsg3408
```
Comment (abbottry) [inline app/assets/javascripts/modules/map/leaflet-plugins/proj.js.coffee]: Delete? Do we just not use this now?

File: app/assets/javascripts/modules/map/leaflet-plugins/proj.js.coffee [hunk 2]
```diff
@@ -3,83 +3,78 @@
...
+        256.0
+      ],
+      bounds: L.Bounds([
+        [-4194304, -4194304],
+        [4194304, 4194304]
+      ])
+    }
+  )
+
+  # # NSIDC EASE-Grid North
```
Comment (abbottry) [inline app/assets/javascripts/modules/map/leaflet-plugins/proj.js.coffee]: I'm cool with leaving this type of stuff around if we have intentions of using it or needing it for reference, is that the case here?

File: spec/features/map/granule_hole_display_spec.rb [hunk 3]
```diff
@@ -14,7 +14,8 @@
       end
 
       it 'draws footprints that contain holes' do
-        expect(page).to have_selector('.leaflet-overlay-pane path', count: 14)
+        # expect(page).to have_selector('.leaflet-overlay-pane path', count: 14)
```
Comment (abbottry) [inline spec/features/map/granule_hole_display_spec.rb]: I know some things changed with layers, but this raises a flag for me.. Is there a known reason these counts are less?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -12,6 +12,10 @@
 Capybara.register_driver :selenium do |app|
   options = Selenium::WebDriver::Chrome::Options.new(
     args: %w(headless disable-gpu no-sandbox --window-size=1440,900 --disable-notifications)
+
+    ### use these args for debugging in chrome.
```
Comment (abbottry) [inline spec/spec_helper.rb]: I think we could just add the `--remote-debugging` permanently to the args above, and the `Open Chrome` I think we should just pop in the README?

File: app/assets/javascripts/modules/map/leaflet-plugins/interpolation.js.coffee [hunk 5]
```diff
@@ -87,21 +87,28 @@ ns.interpolation = do (L, gcInterpolate = window.edsc.map.geoutil.gcInterpolate,
...
 
 
-  # Overrides the default projectLatLngs in Polyline and Polygon to project and interpolate the
+  # Overrides the default projectLatLngs in Polyline to project and interpolate the
   # path instead of just projecting it
-  projectLatlngs = ->
-    interpolated = projectPath(@_map, @_latlngs, @_holes, @_interpolationFn)
-    @_originalPoints = interpolated.boundary
-    @_holePoints = interpolated.holes
+  projectLatlngs = (latlngs, result, projectedBounds) ->
```
Comment (abbottry) [inline app/assets/javascripts/modules/map/leaflet-plugins/interpolation.js.coffee]: Can we get a few comments in here? I'm really lost in this method.

File: app/assets/javascripts/modules/map/leaflet-plugins/gibs_tile_layer.js.coffee [hunk 6]
```diff
@@ -1,34 +1,25 @@
...
-      format: 'jpeg'
-      tileSize: 512
-      extra: ''
-
-    arcticOptions: L.extend({}, parent.arcticOptions, projection: 'EPSG3413', lprojection: 'epsg3413', endpoint: 'arctic')
-    antarcticOptions: L.extend({}, parent.antarcticOptions, projection: 'EPSG3031', lprojection: 'epsg3031', endpoint: 'antarctic')
-    geoOptions: L.extend({}, parent.geoOptions, projection: 'EPSG4326', lprojection: 'epsg4326', endpoint: 'geo')
+  class GibsTileLayer extends L.TileLayer
+    validForProjection: (proj) ->
+      # console.log 'valid?'
```
Comment (trevorlang) [inline app/assets/javascripts/modules/map/leaflet-plugins/gibs_tile_layer.js.coffee]: Stray console log?
```suggestion
```

File: app/assets/javascripts/modules/map/leaflet-plugins/spherical_polygon.js.coffee [hunk 7]
```diff
@@ -44,11 +46,18 @@ ns.L.sphericalPolygon = do (L, geoutil=ns.geoutil, Arc=ns.Arc, Coordinate=ns.Coo
...
         @addLayer(@_boundaries)
 
+    addLatLng: (latlng) ->
+      latlng = L.latLng(latlng)
+      @_latlngs.push(latlng)
+      @_bounds.extend(latlng)
+      @redraw()
+
     getLatLngs: ->
+      console.log 'getLatLngs'
```
Comment (trevorlang) [inline app/assets/javascripts/modules/map/leaflet-plugins/spherical_polygon.js.coffee]: Stray console log?
```suggestion
```

File: app/assets/javascripts/modules/map/map.js.coffee [hunk 8]
```diff
@@ -73,11 +52,24 @@ ns.Map = do (window,
...
       @isMinimap = isMinimap
       $(el).data('map', this)
       @layers = []
-      map = @map = new L.Map(el, zoomControl: false, attributionControl: false)
+
+      map = @map = new L.Map(
+        el,
+        zoomControl: false,
+        attributionControl: false,
+        # turning off zoom animations fixes the fitBounds issue on sticky granules, but looks bad
```
Comment (trevorlang) [inline app/assets/javascripts/modules/map/map.js.coffee]: Are we going to circle back on this? Or live without it for now?

File: app/assets/javascripts/modules/map/map.js.coffee [hunk 9]
```diff
@@ -270,30 +275,21 @@ ns.Map = do (window,
...
         zoom: 0
-        continuousWorld: true
-        noWrap: true
-        worldCopyJump: false
         center: [-90, 0]
       geo:
         crs: ProjExt.epsg4326
         minZoom: 0
-        maxZoom: 7 # This should probably go to 11 when we have higher resolution imagery
+        maxZoom: 8 # This should probably go to 11 when we have higher resolution imagery
```
Comment (trevorlang) [inline app/assets/javascripts/modules/map/map.js.coffee]: https://media.giphy.com/media/lvlLuc2zhi39C/giphy.gif

Comment (macrouch) [reply app/assets/javascripts/modules/map/map.js.coffee]: 🎸

**Pull request merged**